tensorflow-framework/src/main/java/org/tensorflow/framework/data/Dataset.java

```diff
@@ -107,6 +123,72 @@ public final Dataset take(long count) {
         tf, this.getVariant(), tf.constant(count), this.getOutputTypes(), this.getOutputShapes());
   }
 
+  /**
+   * Returns a new Dataset which maps a function across all elements from this iterator, on a single
+   * component of each element.
+   *
+   * <p>For example, suppose each element is a `List<Operand<?>>` with 2 components: (features,
+   * labels).
+   *
+   * <p>Calling `iterator.mapOneComponent(0, features -> tf.math.mul(features, tf.constant(2)))`
+   * will map the function over the `features` component of each element, multiplying each by 2.
+   *
+   * @param index The index of the component to transform.
+   * @param mapper The function to apply to the target component.
+   * @return A new DatasetIterator applying `mapper` to the component at the chosen index.

```

Comment:
Update the return type to `Dataset` instead of `DatasetIterator` (same for other methods below)

Reply:
Thanks, updated this!